doc/sdk-cli-reference.md

```diff
@@ -0,0 +1,177 @@
+# CLI Guide
+
+```bash
+Usage:
+  operator-sdk [command]
+```
+
+## build
+
+### Args
+
+* image - is the container image to be built, e.g. "quay.io/example/operator:v0.0.1". This image will be automatically set in the deployment manifests.
+
+### Flags
+
+* `-h, --help` - help for build
+
+### Use
+
+The operator-sdk build command compiles the code, builds the executables,
+and generates Kubernetes manifests. After build completes, the image would be built locally in docker. Then it needs to be pushed to remote registry.
+
+### Example:
+
+#### Build
+
+```bash
+operator-sdk build quay.io/example/operator:v0.0.1
+
+// Output:
+building example-operator...
+
+building container quay.io/example/operator:v0.0.1...
+Sending build context to Docker daemon  163.9MB
+Step 1/4 : FROM alpine:3.6
+ ---> 77144d8c6bdc
+Step 2/4 : ADD tmp/_output/bin/example-operator /usr/local/bin/example-operator
+ ---> 2ada0d6ca93c
+Step 3/4 : RUN adduser -D example-operator
+ ---> Running in 34b4bb507c14
+Removing intermediate container 34b4bb507c14
+ ---> c671ec1cff03
+Step 4/4 : USER example-operator
+ ---> Running in bd336926317c
+Removing intermediate container bd336926317c
+ ---> d6b58a0fcb8c
+Successfully built d6b58a0fcb8c
+Successfully tagged quay.io/example/operator:v0.0.1
+```
+
+## completion
+
+### Available Commands
+
+#### bash - Generate bash completions
+
+##### Flags
+
+* `-h, --help` - help for bash
+
+#### zsh - Generate zsh completions
+
+##### Flags
+
+* `-h, --help` - help for zsh
+
+### Flags
+
+* `-h, --help` - help for completion
+
+### Use
+
+Generators for shell completions
+
+Example:
+
+```bash
+operator-sdk completion bash
+
+// Output:
+# bash completion for operator-sdk                         -*- shell-script -*-
+...
+# ex: ts=4 sw=4 et filetype=sh
+```
+
+## generate
+
+### Available Commands
+
+#### k8s - Generates Kubernetes code for custom resource
+
+##### Use
+
+k8s generator generates code for custom resource given the API spec
+to comply with kube-API requirements.
+
+##### Flags
+
+* `-h, --help` - help for k8s
+
```

Comment:
Maybe we could add an `Example` section to be consistent

Reply:
Added it...also tried to keep the consistency in the generate olm-catalog doc as well.
Thanks for the suggestion!